### DISCUSSION TITLE
Errors With Declarations

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#28888] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am using the following code:
'''InputParameters
IntegralDirectedSurfaceForceFEM::validParams()
{
  InputParameters params = SideIntegralPostprocessor::validParams();
  params.addClassDescription("Computes the directed force from friction and pressure differences on a surface.");
  params.addRequiredParam<MooseFunctorName>("vel_x", "The velocity in the x-direction.");

  : SideIntegralPostprocessor(parameters),
    _vel_x(getFunctor<Real>("vel_x")),'''

I am getting the following error:
'''/home/twise/projects/crab/src/postprocessors/IntegralDirectedSurfaceForceFEM.C:36:28: error: invalid initialization of reference of type 'const VariableValue&' {aka 'const MooseArray<double>&'} from expression of type 'Moose::Functor<double>' {aka 'const Moose::FunctorEnvelope<double>'}
   36 |     _vel_x(getFunctor<Real>("vel_x")), //  _vel_x(getFunctor<Real>("vel_x")),
      |            ~~~~~~~~~~~~~~~~^~~~~~~~~'''

**Comment by TLWise** [comment]
Adding Templ<is_ad> appears to have eliminated the problem.

**Reply by GiudGiud** [reply, marked as answer]
the error here

> invalid initialization of reference of type 'const VariableValue&' {aka 'const MooseArray&'} from expression of type 'Moose::Functor'

says that you have _vel_x as a variable value, which you can initialize with coupledValue from the Coupleable interface. 
But you are trying to initialize it with getFunctor, which is meant for retrieving references to functors
